File: Linear Shoal/Stack/Q1. Build an Array With Stack Operations/Q1_BuildanArrayWithStackOperations.py
# Solution class
class Solution(object):

    # Function to build the operations list
    def buildArray(self, target, n):

        # List to store operations
        result = []

        # Current number to process
        current = 1  # starts from 1

        # Loop through each number in target
        for num in target:

            # If current number is smaller than target number
            while current < num:

                # Add "Push" and "Pop" operations
                result.extend(["Push", "Pop"])

                # Move to next number
                current += 1

            # When current equals target number, push it
            result.append("Push")

            # Move to next number
            current += 1

        # Return the list of operations
        return result

File: Stack/Q1. Build an Array With Stack Operations/Q1_BuildanArrayWithStackOperations.java
// Solution class
class Solution {

    // Method to build operations array
    public static List<String> buildArray(int[] target, int n) {

        // List to store operations
        List<String> result = new ArrayList<>();

        // Current number to be processed
        int current = 1; // starting from 1

        // Loop through each element in target
        for (int i = 0; i < target.length; i++) {

            // If current number is smaller than target value
            while (current < target[i]) {

                // Push the current number
                result.add("Push");

                // Pop it immediately
                result.add("Pop");

                // Move to next number
                current++;
            }

            // When current equals target value, push it
            result.add("Push");

            // Move to next number
            current++;
        }

        // Return the list of operations
        return result;
    }
}
